### PR TITLE
feat(perf-issues): Update occurrence check to account for group type

### PR DESCRIPTION
**Pull request by udameli**
Add occurrence limits by group type to unblock rollout of new performance issues types (e.g. long span).

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -150,7 +150,12 @@
     **settings.SENTRY_PERFORMANCE_ISSUES_RATE_LIMITER_OPTIONS
 )
 PERFORMANCE_ISSUE_QUOTA = Quota(3600, 60, 5)
+
 GROUPHASH_IGNORE_LIMIT = 3
```
Comment (wedamija) [inline src/sentry/event_manager.py:154]: `GROUPHASH_IGNORE_LIMIT_DEFAULT`?

**Review by wedamija — APPROVED** [review]

File: src/sentry/event_manager.py (hunk 1 above)
Comment (udameli) [reply src/sentry/event_manager.py:154]: ah yes :)

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -2377,14 +2383,15 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
 
 
 @metrics.wraps("performance.performance_issue.should_create_group", sample_rate=1.0)
-def should_create_group(client: Any, grouphash: str) -> bool:
+def should_create_group(client: Any, grouphash: str, type: GroupType) -> bool:
     times_seen = client.incr(f"grouphash:{grouphash}")
     metrics.incr(
         "performance.performance_issue.grouphash_counted",
         tags={"times_seen": times_seen},
```
Comment (rjo100) [inline src/sentry/event_manager.py:2390]: add a tag here for `GroupType` (human readable string of the enum, if possible)

**Review by rjo100 — COMMENTED** [review]
only comment is definitely monitor the graphs for a day or two as the expiration window is "slippery" in that it's not a hard 24 hour window but a soft window that expires 24 hours after the last seen hash

File: src/sentry/event_manager.py (hunk 2 above)
Comment (udameli) [reply src/sentry/event_manager.py:2390]: good idea for sure 👍

Comment (udameli) [reply src/sentry/event_manager.py]: added 👍

**Comment by udameli** [comment]
> only comment is definitely monitor the graphs for a day or two as the expiration window is "slippery" in that it's not a hard 24 hour window but a soft window that expires 24 hours after the last seen hash

that makes sense. I'll keep an eye on it.
depending on the noise levels of new detectors, I might set up a custom (3 days max) window per issue type

**Comment by rjo100** [comment]
> > only comment is definitely monitor the graphs for a day or two as the expiration window is "slippery" in that it's not a hard 24 hour window but a soft window that expires 24 hours after the last seen hash
> 
> that makes sense. I'll keep an eye on it. depending on the noise levels of new detectors, I might set up a custom (3 days max) window per issue type

yeah I just wanted to keep the redis data as simple/minimal as possible so was just storing a counter and let redis do the heavy lifting on expiration. even with this code there's some concurrency stuff going on where counters can spike up well beyond 3 before the key gets deleted

**Pull request merged**